Cargo.toml

```diff
@@ -17,12 +17,12 @@ categories = ["api-bindings", "science"]
 keywords = ["linear-algebra"]
 
 [dependencies]
-libc = "0.2"
+libc = "0.2.84"
 
 [dependencies.num-complex]
-version = "0.2"
+version = "0.3.1"
```

Comment:
I propose to drop `.1`, or is there a reason to keep it?

Reply:
This is not needed indeed since this crate works with num-complex 0.3.0. If you actually want to support all "0.3.x" versions of num-complex and other crates, I will drop all patch versions.

IMO, version specification without patch version, `num-complex = "0.3"` in this case, means this crate works with all versions of num-complex `0.3.x`, but it is generally too strict requirements comparing that `num-complex = "0.3.1"` means this crate works with num-complex `0.3.x` where `x >= 1`.  We usually test these crates with latest dependent crates, and cargo users can get latest crates easily. Thus I encourage to use full version specification like "0.3.1" always to avoid mysterious broken Cargo.toml

Reply:
I think we should rely on semantic versioning here. Before version 1.0, there are no real promises, but people usually follow the convention that patch versions are backward compatible. Specifying a patch version is unnecessary constraint, which one would not want to have unless one knows that something new was added, and that something new is needed. I would stop at minor versions in all dependencies.

Reply:
OK. I will remove them.